frontend/cards/Card.js

```diff
@@ -0,0 +1,217 @@
+import React from 'react'
+import { Text, View, Image, TouchableHighlight, StyleSheet } from 'react-native'
+import AntDesign from 'react-native-vector-icons/AntDesign'
+import Icon from 'react-native-vector-icons/FontAwesome'
+import PropTypes from 'prop-types'
+import imgStyles from '../../styles/cardImage.js'
+import { ID } from 'react-native-dotenv'
+import AsyncStorage from '@react-native-community/async-storage'
+import Alert from '../modals/Alert.js'
+import friendsApi from '../apis/friendsApi.js'
+
+const font = 'CircularStd-Medium'
+const hex = '#F15763'
+var id = ''
+
+export default class Card extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      errorAlert: false,
+      deleteFriend: false,
+      renderOption: this.props.currentUser !== this.props.username,
+      status: this.props.status,
+      pressed: false,
+    }
+    AsyncStorage.getItem(ID).then((res) => {
+      id = res
+    })
+  }
+
+  async deleteFriend() {
+    friendsApi
+      .removeFriendship(id, this.props.id)
+      .then(() => {
+        this.setState({ deleteFriend: false })
+        var filteredArray = this.props.total.filter((item) => {
+          return item.username !== this.props.username
+        })
+        this.props.press(this.props.id, filteredArray, true)
+      })
+      .catch(() => this.setState({ errorAlert: true, deleteFriend: false }))
+  }
+
+  async acceptFriend() {
+    friendsApi
+      .acceptFriendRequest(id, this.props.id)
+      .then(() => {
+        this.setState({ requested: 'friends' })
+      })
+      .catch(() => this.setState({ errorAlert: true }))
+  }
+
+  async addFriend() {
+    friendsApi
+      .createFriendship(id, this.props.id)
+      .then(() => {
+        this.setState({ requested: 'requested' })
+      })
+      .catch(() => this.setState({ errorAlert: true }))
+  }
+
+  async rejectFriend() {
+    friendsApi
+      .removeFriendship(id, this.props.id)
+      .then(() => {
+        this.setState({ requested: 'add' })
+      })
+      .catch(() => this.setState({ errorAlert: true }))
+  }
+
+  render() {
+    return (
+      <View style={styles.container}>
+        {this.props.image.includes('file') || this.props.image.includes('http') ? (
```

Comment:
isha has changed handling of images, see her abt how to do this. looks like the code will conflict quite a bit i think unfortunately